### ISSUE TITLE
OTA Update Disconnected

### ISSUE THREAD

**Issue by hemaltandel1** [timpur/HomieManagement#2]
Hi @timpur ,

Great work... thank you.

I am playing with Homie Management for a couple of day's now. It's awesome. My QTA Update is not working though. I tried to debug it but not able to find the cause of the issue. It's showing Progress: 154323/437280 Bytes and after that, device MQTT Disconnected from MQTT message. I tried several times and it's almost causing an issue almost at the same size. OTA_Update.py is uploading same firmware without any issue. The only difference i have noticed between two firmware is
on OTA_Updater.py ,

Receiving OTA payload
â†• OTA started
Triggering OTA_STARTED event...
**Firmware is binary**

on Homie Management,

Receiving OTA payload
↕ OTA started
Triggering OTA_STARTED event...
**Firmware is base64-encoded**

Don't know if that caused an issue.

John

**Comment by timpur** [comment]
Yes Homie Management uses base64, but this shouldn't be an issue. Might add the option for you to select what encoding type you want to use for the firmware.

Ive been meaning to find time to fix up a lot this this tool, but it should work. Are you using the latest dev ? Try using dev, believe there were some big changes since the v1.0....

Sorry just been busy with homie stuff. There are also new changes with homie 2.1 coming which means this tool needs updating. When i make this change to Homie Management, ill make a new release.

Still things to come, like node view and management, enhancements to firmware, like auto reload and stuff, Add the new capabilities from homie 2.1. Ill make a new issue with this list and you can give me your ideas also.

**Comment by hemaltandel1** [comment]
Great... yes i am using latest dev version. Don't know but it always disconnected at that point. python version didn't failed single time. Will add my thoughts to new issue.

**Comment by timpur** [comment]
I will look into thanks. Sorry its still in its infancy.

**Comment by hemaltandel1** [comment]
@timpur No need to be sorry. You are doing great work.
